### PR TITLE
Feature/ch11

### PR DESCRIPTION
**Pull request by monkeywithbanana**
Chapter 11 code examples

### CODE REVIEW

File: 11/RangeBasedForLoopExample.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		nums.push_back(5);
+
+		for (int n : nums)
+		{
+			n++;
+		}
+
+		cout << "Print nums:" << endl;
+
+		for (auto it = nums.begin(); it != nums.end(); ++it)
```
Comment (popekim) [inline 11/RangeBasedForLoopExample.cpp:27]: u can print with range-based for as well

File: 11/RangeBasedForLoopExample.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		}
+
+		for (int& n : nums)
+		{
+			n--;
+		}
+
+		cout << "Print nums again:" << endl;
+
+		for (auto it = nums.begin(); it != nums.end(); ++it)
```
Comment (popekim) [inline 11/RangeBasedForLoopExample.cpp:39]: same

File: 11/SpeedTestExample.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+		unorderedSet.reserve(MAX_NUMBER_IN_SET);
+
+		static_assert(MAX_NUMBER_IN_SET > NUMBER_TO_INSERT, "MAX_NUMBER_IN_SET should be greater than NUMBER_TO_INSERT");
+
+		for (int i = 0; i < MAX_NUMBER_IN_SET; i++)
+		{
+			if (i == NUMBER_TO_INSERT)
+			{
+				continue;
```
Comment (popekim) [inline 11/SpeedTestExample.cpp:30]: continue? what does it do? it only skips one number

Comment (popekim) [reply 11/SpeedTestExample.cpp:30]: NUMBER_TO_INSERT later i guess

File: 11/UsingUniquePointersExample.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+		unique_ptr<MyVector> myVector = make_unique<MyVector>(3, 5);
+
+		cout << "X: " << myVector->GetX() << ", Y: " << myVector->GetY() << endl;
+
+		// Compiler error
+		// unique_ptr<MyVector> copyOfMyVector1 = myVector;
+		// unique_ptr<MyVector> copyOfMyVector2(myVector);
+
+		myVector.reset(new MyVector(1, 5));
```
Comment (popekim) [inline 11/UsingUniquePointersExample.cpp:26]: hmm not sure if it's possibel but can we make vector with make_unique and wrap it with std::move?

**Review by popekim — APPROVED** [review]
merge after you finish

File: 11/RangeBasedForLoopExample.cpp (hunk 1 above)
Comment (monkeywithbanana) [reply 11/RangeBasedForLoopExample.cpp:27]: I don't know... This iterator loop (and the one below) is to prove the difference between `for (int n : nums)` and `for (int& n : nums)` using the loop that the students already know how. If I use range-based here, then it kinda feels like... I'm trying to prove a new concept using the new concept, which is kinda weird? no?

File: 11/SpeedTestExample.cpp (hunk 3 above)
Comment (monkeywithbanana) [reply 11/SpeedTestExample.cpp:30]: I'll rename

File: 11/UsingUniquePointersExample.cpp (hunk 4 above)
Comment (monkeywithbanana) [reply 11/UsingUniquePointersExample.cpp:26]: Like this? 

```
unique_ptr<char> char3(move(make_unique<float>(1.0f)));
```

This is compile error

**Pull request merged**